.github/workflows/release.yml

```
@@ -247,11 +248,13 @@ jobs:
 
   deploy-web:
+    needs: release-please
     runs-on: ubuntu-latest
-    # Deploy apps/web to Cloudflare Pages on every merge to main.
-    # Skipped on workflow_dispatch (which is reserved for npm re-publish only).
+    # Deploy apps/web to Cloudflare Pages when a project release is created.
+    # Skipped when no release happened (e.g. mid-cycle pushes) or when
+    # Cloudflare secrets are not yet configured.
     # Requires CLOUDFLARE_API_TOKEN and CLOUDFLARE_ACCOUNT_ID secrets.
-    if: github.event_name == 'push' && secrets.CLOUDFLARE_API_TOKEN != ''
+    if: needs.release-please.outputs.releases_created == 'true' && secrets.CLOUDFLARE_API_TOKEN != ''
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)